test/integration_test.py

```diff
@@ -1035,3 +1044,103 @@ def test_local_filtered_index_endpoints(self):
+                             output_dir=d)
+
+            def file_name(part):
+                return f'{fqid.uuid}.{part}.json'
+
+            manifest_file_name = file_name('manifest')
+            metadata_file_name = file_name('metadata')
+            expected_files = sorted([manifest_file_name, metadata_file_name])
+            generated_files = sorted(os.listdir(d))
+            self.assertListEqual(generated_files, expected_files)
+
+            with open(f'{d}/{manifest_file_name}') as f:
+                manifest = json.load(f)
+            with open(f'{d}/{metadata_file_name}') as f:
+                metadata = json.load(f)
+
+            self.assertIsInstance(manifest, list)
+            self.assertIsInstance(metadata, dict)
+
+            manifest_files = sorted(e['name'] for e in manifest if e['indexed'])
+            metadata_files = sorted(metadata.keys())
+
+            if catalog.plugins['repository'] == 'canned':
+                # FIXME: Manifest entry not generated for links.json by
+                #        StagingArea.get_bundle
+                #        https://github.com/DataBiosphere/hca-metadata-api/issues/52
+                assert 'links.json' not in manifest_files
+                metadata_files.remove('links.json')
+
+            self.assertListEqual(manifest_files, metadata_files)
+
+    def test_can_bundle_configured_catalogs(self):
+        for catalog_name, catalog in config.catalogs.items():
+            if catalog.is_integration_test_catalog:
+                with self.subTest(catalog=catalog.name,
+                                  repository=catalog.plugins['repository']):
+                    self._test_catalog(catalog)
+
+    canned_repo = 'https://github.com/HumanCellAtlas/schema-test-data/tree/master/tests'
+
+    @mock.patch.dict(os.environ,
+                     azul_canned_sources=f'{canned_repo}:')
+    @mock.patch.object(azul.Config, 'catalogs', new_callable=PropertyMock)
+    # Must be mocked independently due to caching
+    @mock.patch.object(azul.Config, 'default_catalog', new_callable=PropertyMock)
+    def test_can_bundle_canned_repository(self, mock_default_catalog, mock_catalogs):
+        mock_catalog_name = 'testcanned'
+        mock_catalog = config.Catalog(name=mock_catalog_name,
+                                      atlas='hca',
+                                      plugins={
+                                          'metadata': 'hca',
+                                          'repository': 'canned'
+                                      })
+        mock_default_catalog.return_value = mock_catalog_name
+        mock_catalogs.return_value = {mock_catalog_name: mock_catalog}
+        self._test_catalog(mock_catalog)
+
+    def bundle_fqid(self, catalog: CatalogName) -> SourcedBundleFQID:
+        sources = list(self.azul_client.repository_plugin(catalog).sources)
+        random.shuffle(sources)
+        prefix = self.random_prefix(length=4)
+        while True:
```

Comment:
This is strikingly similar to _prepare_notifications and _prune_test_bundles(max_bundles=1). Please try to consolidate. If the consolidation requires refactoring those two methods (I presume it does and I think extracting a superclass is the appropriate type of refactoring) do that in a separate, first commit. 

Whenever we use randomness in a test, there must be a way to reproduce a particular outcome by seeing the random generator. The code in _prepare_notifications & _prune_test_bundles does that. You version does not.